api/src/main/java/org/openmrs/module/fgh/mpi/MpiHttpClient.java

```diff
@@ -122,19 +168,110 @@ public void submitPatient(String patientData) throws Exception {
 			log.debug("Patient data -> " + patientData);
 		}
 		
-		List<Map<String, Object>> mpiIdsResp = submitRequest(SUBPATH_PATIENT, patientData, List.class);
+		AdministrationService adminService = Context.getAdministrationService();
 		
-		if (log.isDebugEnabled()) {
-			log.debug("MPI patient submission response: " + mpiIdsResp);
-		}
+		String mpiSystem = adminService.getGlobalProperty(GP_MPI_SYSTEM);
 		
-		if (CollectionUtils.isEmpty(mpiIdsResp) || mpiIdsResp.get(0) == null) {
-			throw new APIException("An empty response was received when the patient was submitted");
+		if (MpiUtils.isSanteMPI(mpiSystem)) {
+			Map<String, Object> santeResponse = submitRequest(SUBPATH_PATIENT, patientData, Map.class);
+			
+			if (log.isDebugEnabled()) {
+				log.debug("MPI patient submission response: " + santeResponse);
+			}
+			
+			if (santeResponse == null || santeResponse.isEmpty()) {
```

Comment:
You can merge these into a single check with MapUtils.isEmpty()

Reply:
Ok